deals-add.js

```diff
@@ -0,0 +1,223 @@
+	resetFeed();
+	currentCategory = 'Daily'
+	getDeals(currentCategory);
+
+});
+
+document.getElementsByClassName("price-drop-button")[0].addEventListener("click", function() {
+	resetFeed();
+	currentCategory = 'PriceDrop'
+	getDeals(currentCategory);
+});
+
+document.getElementsByClassName("popular-button")[0].addEventListener("click", function() {
+	resetFeed();
+	currentCategory = 'Popular'
+	getDeals(currentCategory);
+});
+
+// creates a new product card
+function createNewCard() {
+	var newCard = dealsContainer.firstElementChild.cloneNode(true);
+	dealsContainer.appendChild(newCard);
+}
+
+// displays the correct currency format
+function getCurrency(deal) {
+	if (deal.listing.currency == "SGD") {
+		return "S$";
+	}
+}
+
+// rounds the currency to the specified decimal places
+function round(value, decimals) {
+  var savings = Number(Math.round(value+'e'+decimals)+'e-'+decimals);
+  if (savings % 1.00 > 0) { // if there are decimals, display with 2 decimal places
+		return savings.toFixed(2);
+	}
+	else { // if there are no decimals, do not display decimal places
+		return savings;
+	}
+}
+
+// calculate amount of savings from previousPrice and currentPrice
+function calculateSavings(deal) {
+	var savings = parseFloat(deal.listing.previousPrice) - parseFloat(deal.listing.currentPrice);
+	return round(savings, 2); // more accurate rounding method
+}
+
+// add ellipsis where product name cuts off
+function clamp(dealTitle) {
+	$clamp(dealTitle, {clamp: 2});
+}
+
+/* can be used for activities feed in future
+function getTimeAgo(lastCreated) {
+	var date = lastCreated.substring(0, 10);
+	date = date.replace(/-/g, "");
+	date = date + ", " + lastCreated.substring(11, 16);
+
+	return moment(date, "YYYYMMDD, hh:mm").fromNow();
+
+}*/
+
+// change the string to title case
+function toTitleCase(str)
+{
+     return str.replace(/\w\S*/g, 
+    	function(txt){
+    		return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
+    	});
+}
+
+// removes all existing card except the first one, hides the first one
+function resetFeed() {
+	offset = 0;
+	hasMore = true;
+	dealsContainer.firstChild.style.visibility='hidden';
+	var j = 0;
+	for (j = document.getElementsByClassName("deal-card").length; j > 1; j--) {
+		dealsContainer.removeChild(dealsContainer.lastChild);
+	}
+}
+
+// do API call, populate cards with deal info
+function getDeals(filter) {
+	// sets isFetching to true to prevent multiple triggers
+	isFetching = true;
+	$.ajax ({
+		method: 'GET',
+		url: 'https://ratex.co/store/api/products' + '?filter=' + filter
+	})
+	.done(function (response) {
+		// sets address bar with parameters
+		window.history.pushState({urlPath:'/trending-deals-rates?category=' + currentCategory},"",'/trending-deals-rates?category=' + currentCategory);
+		
+		dealsContainer.firstChild.style.visibility='visible';
+		
+		var startOfData = 0;
+		for(i = offset; startOfData < response.data.length && hasMore; i++, startOfData++) {
+			if (i !== startOfData) {
```

Comment:
I think the whole chunk inside the `for loop` can be refactored out to a separate function. This will improve readability